### PR TITLE
Add method appendHex to StringBuilder

### PR DESCRIPTION
**Pull request by wenshao**
There are a lot of append(Integer.toHexString(x)) codes in the JDK. Adding methods StringBuilder#appendHex(int) and StringBuilder.appendHex(long) can reduce these repeated codes. This method is not only useful internally to the JDK, but also to users.

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [ ] Commit message must refer to an issue

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15998/head:pull/15998` \
`$ git checkout pull/15998`

Update a local copy of the PR: \
`$ git checkout pull/15998` \
`$ git pull https://git.openjdk.org/jdk.git pull/15998/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15998`

View PR using the GUI difftool: \
`$ git pr show -t 15998`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15998.diff">https://git.openjdk.org/jdk/pull/15998.diff</a>

</details>

### CODE REVIEW

**Comment by AlanBateman** [comment]
Step 2 in the Developer's Guide is "Socialize your change" [1] where it advises that API proposals should be discussed on the mailing list. Please do that as it's no helpful to start with a PR/code when discussing APIs.

[1] https://openjdk.org/guide/#socialize-your-change

File: src/java.base/share/classes/java/net/Inet6Address.java [hunk 1]
```diff
@@ -917,8 +917,8 @@ public boolean isIPv4CompatibleAddress() {
     static String numericToTextFormat(byte[] src) {
         StringBuilder sb = new StringBuilder(39);
         for (int i = 0; i < (INADDRSZ / INT16SZ); i++) {
-            sb.append(Integer.toHexString(((src[i<<1]<<8) & 0xff00)
-                                          | (src[(i<<1)+1] & 0xff)));
+            sb.appendHex(((src[i<<1]<<8) & 0xff00)
+                                          | (src[(i<<1)+1] & 0xff));
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/net/Inet6Address.java:921]: **Nit:**
```suggestion
            sb.appendHex(((src[i<<1]<<8) & 0xff00)
                         | (src[(i<<1)+1] & 0xff));
```

**Comment by wenshao** [comment]
Using the new method, the performance is improved by approximately 26% when running on MaxBookPro M1 Max.

```
Benchmark                         Mode  Cnt   Score   Error  Units
StringBuilders.appendHex8         avgt   15  15.259 ? 0.121  ns/op (+26.19%)
StringBuilders.appendHexCombine8  avgt   15  19.256 ? 0.059  ns/op
```

The test code is as follows:
```java
public class StringBuilders {
    @Benchmark
    public String appendHexCombine8() {
        StringBuilder result = new StringBuilder();
        result.append(Integer.toHexString(2048));
        result.append(Integer.toHexString(31337));
        result.append(Integer.toHexString(0xbeefcace));
        result.append(Integer.toHexString(9000));
        result.append(Integer.toHexString(4711));
        result.append(Integer.toHexString(1337));
        result.append(Integer.toHexString(2100));
        result.append(Integer.toHexString(2600));
        return result.toString();
    }
    @Benchmark
    public String appendHex8() {
        StringBuilder result = new StringBuilder();
        result.appendHex(2048);
        result.appendHex(31337);
        result.appendHex(0xbeefcace);
        result.appendHex(9000);
        result.appendHex(4711);
        result.appendHex(1337);
        result.appendHex(2100);
        result.appendHex(2600);
        return result.toString();
    }
}
```

**Comment by wenshao** [comment]
According to the email discussion, it is not recommended to add this type of API.

**Pull request closed without merging**